internal/bminventory/inventory.go

```diff
@@ -2978,6 +2978,11 @@ func (b *bareMetalInventory) V2DeregisterHostInternal(ctx context.Context, param
 		hostutil.GetHostnameForMsg(&h.Host))
 
 	if h.ClusterID != nil {
+		if interactivity == Interactive {
+			if _, err = b.refreshClusterStatus(ctx, h.ClusterID, b.db); err != nil {
+				log.WithError(err).Warnf("Failed to refresh cluster after de-registerating host <%s>", params.HostID)
```

Comment:
Only log ? Why not return error ?

Reply:
Per discussion with @avishayt we decided to fail gracefully in such case (which should be rare).